### PR TITLE
update to use Uint8arrays when necessary

### PR DESCRIPTION
**Pull request by zo-el**

### CODE REVIEW

File: src/store/elementalChat.js [hunk 1]
```diff
@@ -265,9 +274,13 @@ export default {
...
           }
 
-          const messages = [...result.messages]
+          let messages = [...result.messages]
 
           messages.sort((a, b) => a.createdAt[0] - b.createdAt[0])
+          messages = messages.map((msg) => {
+            msg.createdBy = toUint8Array(msg.createdBy)
+            return msg
+          })
```
Comment (timotree3) [inline src/store/elementalChat.js:283]: If this is already mutating the messages I don't you need to use map
```suggestion
          messages = messages.forEach((msg) => {
            msg.createdBy = toUint8Array(msg.createdBy)
          })
```

File: src/store/elementalChat.js [hunk 2]
```diff
@@ -226,13 +226,22 @@ export default {
       message.createdBy = toUint8Array(message.createdBy)
       const channel = payload.channel
       channel.info.created_by = toUint8Array(channel.info.created_by)
-
+      console.log("ACTIVE CHATTER:", channel);
```
Comment (robbiecarlton) [inline src/store/elementalChat.js:229]: Do we still need this log?

**Review by robbiecarlton — APPROVED** [review]
I think this is good for EC. 

But it has me thinking: I imagine this kind of conversion is something we're going to be doing a lot of, and it would be nice to have a more general solution. In a bigger app this will get very noisy and error prone.

File: src/store/elementalChat.js [hunk 3]
```diff
@@ -226,13 +226,22 @@ export default {
...
+      channel.activeChatters = channel.activeChatters.map((c)=>toUint8Array(c))
+      channel.messages = channel.messages.map((msg) => {
+        msg.createdBy = toUint8Array(msg.createdBy)
+        msg.entryHash = toUint8Array(msg.entryHash)
+        msg.createdBy = toUint8Array(msg.createdBy)
+        return msg
+      })
+      const chatters = payload.channel.activeChatters.map((c)=>toUint8Array(c))
+      console.log("Chatter:", chatters);
```
Comment (robbiecarlton) [inline src/store/elementalChat.js:238]: or this one?
The list of chatters is already logged in the zome call args

**Comment by timotree3** [comment]
Why do we need to convert `Buffer`s to `Uint8Array`s at all again? Doesn't MessagePack handle buffers fine?

**Comment by alastairong** [comment]
Signals seems to work fine with this btw

**Pull request merged**